Port to Python3, GTK3 and other fixes [GCI]

srevinsaju opened the pull request:
Used some of @yashagrawal3 's commits on Port To GTK3

yell0wfl4sh committed:
updated utils

yell0wfl4sh committed:
Port to gtk3

yell0wfl4sh committed:
Gtk3 port update

yell0wfl4sh committed:
add and fix toolbar

yell0wfl4sh committed:
fix next and prev button

yell0wfl4sh committed:
remove extra code

srevinsaju committed:
Update upstream changes to WIP port

srevinsaju committed:
Port to Python3 (2to3)

srevinsaju committed:
Fix svg_loader

srevinsaju committed:
Update sugar-activity to sugar-activity3 in activity.info

srevinsaju committed:
Port setup.py to GTK3

srevinsaju committed:
Fix tagplay referring to invaid data

srevinsaju committed:
fix typo from get_propretry to get_property

srevinsaju committed:
Replace ''' to

quozl commented on ICanReadActivity.py:
```diff
@@ -143,25 +143,33 @@ def __init__(self, handle):
-
-        self._images_path = self._lessons_path.replace('lessons', 'images')
-        self._sounds_path = self._lessons_path.replace('lessons', 'sounds')
+#        if os.path.exists(os.path.join('~', 'Activities', 'ICanRead.activity')):
```
Remove code the commented out.

quozl commented on ICanReadActivity.py:
```diff
@@ -184,57 +192,33 @@ def __init__(self, handle):
+        toolbox.toolbar.insert(lesson_toolbar_button, -1)
+        record_toolbar_button.show()
+        toolbox.toolbar.insert(record_toolbar_button, -1)
+        primary_toolbar = toolbox.toolbar
```
Nice.

srevinsaju replied:
@quozl, I didn't understand

quozl replied:
Means _well done_.  Good.

quozl commented on ICanReadActivity.py:
```diff
@@ -305,18 +289,16 @@ def _setup_toolbars(self):
-            stop_button.show()
+        _separator_factory(toolbox.toolbar, False, True)
+        stop_button = StopButton(self)
+        stop_button.props.accelerator = '<Ctrl>q'
```
Accelerator is defined already in StopButton class.

quozl commented on page.py:
```diff
@@ -133,9 +133,9 @@ def page_list(self):
-        self.invalt(0, 0, self._width, self._height)
+        self.cr.rectangle(0, 0, self._width, int(self._height * 2.75))
+        self.cr.fill()
+        # self.invalt(0, 0, self._width, self._height)
```
Remove commented code.

quozl commented on page.py:
```diff
@@ -284,11 +285,11 @@ def _load_card(self):
-        self.invalt(0, 0, self._width, int(self._height * 2.5))
+        self.cr.rectangle(0, 0, self._width, int(self._height * 2.75))
+        self.cr.fill()
+        # self.invalt(0, 0, self._width, int(self._height * 2.5))
```
Remove commented code.

quozl commented on page.py:
```diff
@@ -352,17 +353,17 @@ def read(self):
-        self.invalt(0, 0, self._width, self._height)
+        self.cr.rectangle(0, 0, self._width, int(self._height * 2.75))
+        self.cr.fill()
+        # self.invalt(0, 0, self._width, self._height)
```
Ditto.

quozl commented on page.py:
```diff
@@ -374,10 +375,9 @@ def read(self):
-        self.invalt(0, 0, self._width, self._height)
+        self.cr.rectangle(0, 0, self._width, int(self._height * 2.75))
+        self.cr.fill()
+        # self.invalt(0, 0, self._width, self._height)
```
Ditto.

quozl commented on page.py:
```diff
@@ -482,12 +482,11 @@ def _draw_a_word(self, word, canvas, gc):
-        self.invalt(x, y, w, h)
+        cr.rectangle(0, 0, int(x), int(y))
+        cr.fill()
+        # self.invalt(x, y, bounds.width, bounds.height)
```
Ditto.

quozl commented on page.py:
```diff
@@ -629,8 +612,11 @@ def svg_str_to_pixbuf(svg_string):
     except:
         _logger.debug('failed to load %s', file_path)
         return None
+
```
Strange extra blank lines at end of file.

quozl commented on utils/gplay.py:
```diff
@@ -22,122 +22,154 @@
         if not self.player:
-            # lazy init the player so that videowidget is realized
-            # and has a valid widget allocation
+            # Lazy init the player so that videowidget is realized
```
Nice.

quozl commented on utils/grecord.py:
```diff
@@ -41,170 +46,213 @@ def __init__(self, parent):
+        self._audiopos = 0
+        self._audioline.set_state(Gst.State.PLAYING)
+
+    def transcoding_complete(self):
```
Separate transcoding step was removed from the Record activity, so if you want to remove it here compare against those sources.  It was once needed because hardware could not record and transcode at the same time.

quozl commented on utils/grecord.py:
```diff
@@ -41,170 +46,213 @@ def __init__(self, parent):
-        oggFilepath = os.path.join(self._activity.datapath, 'output.ogg')
-        os.remove( wavFilepath )
-        return False
+        _logger.debug('skip clean up of output.wav')
```
Remove commented code and logger call if not needed.

srevinsaju committed:
Cleanup code and fix GTK Regressions

Transcoding was done to help sugar desktops in wotking with both recording and playing alternatively
This code was no longer neceassary and hence removed.
Redundant comments were removed, and code which was no longer used were also removed (python3)

chimosky commented:
How did you test this? Some of the buttons seem to do nothing and there's nothing in the logs.

srevinsaju commented:
Yup, it didn't work in the python2 branch also before porting. DId you try testing the current master. I have no clue what it does, it (on purpose) shows only a blank screen allows us to record and replay. If you found something else, I would be able to reproduce. Thanks @chimosky

chimosky commented:
Yes I tested master. The [help](https://help.sugarlabs.org/icanread.html) page shows that sounds should be produced when letters are added and no sounds are produced currently.

srevinsaju commented:
@chimosky , I couldn't make it work on any system. I tried the hybrid build and Fedora 18 SOAS. Fedora SOAS's yum was outdated, (probably because its no longer maintained), I can't use git or rsync on Fedora 18, and can't download them too. Due to technical issues, I have abandoned the task temporarily.

quozl commented:
Tested _master_ on Ubuntu 16.04 with Sugar 0.112; failed; the canvas is not drawn, adn the toolbar is offset to the right.  In this screenshot the Terminal activity canvas remains visible, as does the left two icons of the Terminal toolbar.

![Screenshot_ubuntu-xenial_2020-01-22_15:07:11](https://user-images.githubusercontent.com/2936481/72864664-e6e56180-3d28-11ea-8bc4-dfbd75651543.png)

Tested v3 on OLPC XO-4 with OLPC OS 13.2.10, aka Fedora 18 plus Sugar 0.112; passed; the canvas was drawn properly.  It looks just like the [help](https://help.sugarlabs.org/icanread.html).

![Screenshot of _I Can Read Activity_](https://user-images.githubusercontent.com/2936481/72864705-0aa8a780-3d29-11ea-88d6-e152eb702e1c.png)

It is best to get the activity working on GTK 2 before porting it to GTK 3 or Python 3, but there's no reason why the problem can't be fixed on the ported code; just unusual.

My guess is that the fault will be in the widgets in the activity initialisation.

Fedora 18 package archives have moved and OLPC OS 13.2.10 does use http://archives.fedoraproject.org/pub/archive/fedora/linux/releases/18/Everything/i386/os/ instead.  See `/etc/yum.repoos.d`.
